app/screens/browse_channels/channel_dropdown.tsx

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {useIntl} from 'react-intl';
+import {View, Text} from 'react-native';
+
+import CompassIcon from '@components/compass_icon';
+import {ITEM_HEIGHT} from '@components/slide_up_panel_item';
+import Screens from '@constants/screens';
+import {useTheme} from '@context/theme';
+import {TITLE_HEIGHT} from '@screens/bottom_sheet/content';
+import {showModalOverCurrentContext} from '@screens/navigation';
+import {makeStyleSheetFromTheme} from '@utils/theme';
+import {typography} from '@utils/typography';
+
+import {PUBLIC, SHARED} from './constants';
+import DropdownSlideup from './dropdown_slideup';
+
+type Props = {
+    typeOfChannels: string;
+    onPress: (channelType: string) => void;
+    canShowArchivedChannels: boolean;
+    sharedChannelsEnabled: boolean;
+}
+
+const getStyleFromTheme = makeStyleSheetFromTheme((theme: Theme) => {
+    return {
+        channelDropdown: {
+            ...typography('Body', 100, 'SemiBold'),
+            lineHeight: 20,
+            color: theme.centerChannelColor,
+            marginLeft: 20,
+            marginTop: 12,
+            marginBottom: 4,
+        },
+        channelDropdownIcon: {
+            color: theme.centerChannelColor,
+        },
+    };
+});
+
+const BOTTOM_SHEET_HEIGHT_BASE = 55; // Magic number
+
+export function ChannelDropdown({
+    typeOfChannels,
+    onPress,
+    canShowArchivedChannels,
+    sharedChannelsEnabled,
+}: Props) {
+    const intl = useIntl();
+    const theme = useTheme();
+    const style = getStyleFromTheme(theme);
+
+    // Depends on all props, so no need to use a callback.
+    const handleDropdownClick = () => {
+        const renderContent = () => {
+            return (
+                <DropdownSlideup
+                    canShowArchivedChannels={canShowArchivedChannels}
+                    onPress={onPress}
+                    sharedChannelsEnabled={sharedChannelsEnabled}
+                    selected={typeOfChannels}
+                />
+            );
+        };
+
+        let items = 1;
+        if (canShowArchivedChannels) {
+            items += 1;
+        }
+        if (sharedChannelsEnabled) {
+            items += 1;
+        }
+
+        showModalOverCurrentContext(Screens.BOTTOM_SHEET, {
+            renderContent,
+            snapPoints: [(items * ITEM_HEIGHT) + TITLE_HEIGHT + BOTTOM_SHEET_HEIGHT_BASE, 10],
```

Comment:
do we want this bottom sheet to have a snap point near the top of the screen? I think this can have only three options at most, so having the snap point `10` seems unnecessary